langchain_tools_demo/orchestrator/langchain_tools/langchain_tools_orchestrator.py

```diff
@@ -0,0 +1,234 @@
+
+    def user_session_exist(self, uuid: str) -> bool:
+        return uuid in self._user_sessions
+
+    async def user_session_create(self, session: dict[str, Any]):
+        """Create and load an agent executor with tools and LLM."""
+        print("Initializing agent..")
+        if "uuid" not in session:
+            session["uuid"] = str(uuid.uuid4())
+        id = session["uuid"]
+        if "history" not in session:
+            session["history"] = [BASE_HISTORY]
+        history = self.parse_messages(session["history"])
+        client = await self.create_client_session()
+        tools = await initialize_tools(client)
+        prompt = self.create_prompt_template(tools)
+        agent = UserAgent.initialize_agent(client, tools, history, prompt)
+        self._user_sessions[id] = agent
+
+    async def user_session_invoke(self, uuid: str, prompt: str) -> str:
+        user_session = self.get_user_session(uuid)
+        # Send prompt to LLM
+        response = await user_session.invoke(prompt)
+        return response["output"]
+
+    async def user_session_reset(self, uuid: str):
+        user_session = self.get_user_session(uuid)
+        await user_session.close()
+        del user_session
+
+    def get_user_session(self, uuid: str) -> UserAgent:
+        return self._user_sessions[uuid]
+
+    async def get_connector(self) -> TCPConnector:
+        if self.connector is None:
+            self.connector = TCPConnector(limit=100)
+        return self.connector
+
+    async def create_client_session(self) -> ClientSession:
+        return ClientSession(
+            connector=await self.get_connector(),
+            connector_owner=False,
+            headers={},
+            raise_for_status=True,
+        )
+
+    def create_prompt_template(self, tools: List[StructuredTool]) -> ChatPromptTemplate:
+        # Create new prompt template
+        tool_strings = "\n".join(
+            [f"> {tool.name}: {tool.description}" for tool in tools]
+        )
+        tool_names = ", ".join([tool.name for tool in tools])
+        format_instructions = FORMAT_INSTRUCTIONS.format(
+            tool_names=tool_names,
+        )
+        today_date = date.today().strftime("%Y-%m-%d")
+        today = f"Today is {today_date}."
+        template = "\n\n".join(
+            [PREFIX, tool_strings, format_instructions, SUFFIX, today]
+        )
+        human_message_template = "{input}\n\n{agent_scratchpad}"
+        prompt = ChatPromptTemplate.from_messages(
+            [("system", template), ("human", human_message_template)]
+        )
+        return prompt
+
+    def parse_messages(self, datas: List[Any]) -> List[BaseMessage]:
+        messages: List[BaseMessage] = []
+        for data in datas:
+            if data["type"] == "human":
+                messages.append(HumanMessage(content=data["data"]["content"]))
+            if data["type"] == "ai":
+                messages.append(AIMessage(content=data["data"]["content"]))
```

Comment:
It's almost always a bad idea to just ignore bad data. Log or return an error instead so their is some feedback someone did something wrong. 
```suggestion
            if data["type"] == "human":
                messages.append(HumanMessage(content=data["data"]["content"]))
            elif data["type"] == "ai":
                messages.append(AIMessage(content=data["data"]["content"]))
            else:
                raise Exception()
```